src/test/java/api/requests/RequestsAPICreationTests.java

```diff
@@ -1866,6 +1890,76 @@ public void recallNoticeToLoanOwnerIsSentForMovedRecallIfDueDateIsNotChanged() {
     assertThatPublishedLogRecordEventsAreValid();
   }
 
+  @Test
+  public void shouldNotCreateRequestWhenItemRequesterAndPickupServicePointAreNotProvided() {
+    Response postResponse = requestsClient.attemptCreate(new RequestBuilder()
+      .recall()
+      .withItemId(null)
+      .withRequesterId(null)
+      .withPickupServicePointId(null));
+
+    assertThat(postResponse, hasStatus(HTTP_UNPROCESSABLE_ENTITY));
+
+    final JsonObject responseJson = postResponse.getJson();
+
+    assertThat(responseJson, hasErrors(3));
+
+    assertThat(responseJson, hasErrorWith(allOf(
+        hasMessage("Cannot create a request with no item ID"),
+        hasNullParameter("itemId"))));
+
+    assertThat(responseJson, hasErrorWith(allOf(
+      hasMessage("A valid user is required. User is null."),
+      hasNullParameter("userId"))));
+
+    assertThat(responseJson, hasErrorWith(
+      hasMessage("Hold Shelf Fulfillment Requests require a Pickup Service Point")));
+  }
+
+  @Test
+  public void shouldReturnMultipleErrorsWhenRequestContainsNonExistentIds() {
+    final UUID itemId = UUID.randomUUID();
+    final UUID userId = UUID.randomUUID();
+    final UUID pickupServicePointId = UUID.randomUUID();
+
+    Response postResponse = requestsClient.attemptCreate(new RequestBuilder()
+      .recall()
+      .withItemId(itemId)
+      .withRequesterId(userId)
+      .withPickupServicePointId(pickupServicePointId));
+
+    assertThat(postResponse, hasStatus(HTTP_UNPROCESSABLE_ENTITY));
+
+    final JsonObject responseJson = postResponse.getJson();
+
+    assertThat(responseJson, hasErrors(3));
+
+    assertThat(responseJson, hasErrorWith(allOf(
+      hasMessage("Item does not exist"),
+      hasUUIDParameter("itemId", itemId))));
+
+    assertThat(responseJson, hasErrorWith(allOf(
+      hasMessage("A valid user is required. User is null."),
+      hasUUIDParameter("userId", userId))));
+
+    assertThat(responseJson, hasErrorWith(allOf(
+      hasMessage("Pickup service point does not exist"),
+      hasUUIDParameter("pickupServicePointId", pickupServicePointId))));
+  }
+
+  @Test
+  public void shouldReturnFirstNonValidationErrorIfAny() {
+    Response postResponse = requestsClient.attemptCreate(new RequestBuilder()
+      .recall()
+      .withStatus("fake")             // 400
+      .withItemId(null)               // 422
+      .withRequesterId(null)          // 422
+      .withPickupServicePointId(null) // 422
+    );
+
+    assertThat(postResponse, hasStatus(HTTP_BAD_REQUEST));
+  }
```

Comment:
What does this test demonstrate? That a fatal error stops the process?

Reply:
It does not stop the process immediately, but is accumulated along with validation errors. But when we build the response, these non-validation errors get special treatment:
- if all the errors we have accumulated are validation-errors, we return all of them in an `"errors"` array
- if among accumulated errors we have non-validation errors, we return just the first one of them and discard the rest

The second scenario was necessary in order not to break existing clients.

Reply:
Hmm, if the system discards all of the accumulated validation errors in the presence of a non-validation error. What is the value of continuing the process and accumulation when one of those non-validation errors occurs?

Reply:
Irrelevant, as this test no longer exists.